### ISSUE TITLE
lxd init preseed autogeneration (feature request, question)

### ISSUE THREAD

**Issue by cryptofuture** [lxc/lxd#3418]
I'm trying to automate with salt my lxd default setup with `lxd init` right now. And its easier for me to setup lxd just with lxc commands (lxc storage create default btrfs ... etc). However since, we have preseed, maybe I should stick with it.
And my usual (base) configuration is quite common.
```
lxd init --verbose
INFO[06-15|14:10:00] Connecting to a local LXD over a Unix socket 
INFO[06-15|14:10:00] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0
INFO[06-15|14:10:00] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/storage-pools
INFO[06-15|14:10:00] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0
INFO[06-15|14:10:00] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/profiles/default
Do you want to configure a new storage pool (yes/no) [default=yes]? 
Name of the new storage pool [default=default]: 
Name of the storage backend to use (dir, btrfs) [default=dir]: btrfs
Create a new BTRFS pool (yes/no) [default=yes]? 
Would you like to use an existing block device (yes/no) [default=no]? 
Would you like to create a new subvolume for the BTRFS storage pool (yes/no) [default=yes]: 
Would you like LXD to be available over the network (yes/no) [default=no]? 
Would you like stale cached images to be updated automatically (yes/no) [default=yes]? 
Would you like to create a new network bridge (yes/no) [default=yes]? 
What should the new bridge be called [default=lxdbr0]? 
INFO[06-15|14:10:26] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/networks/lxdbr0
What IPv4 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? 
What IPv6 address should be used (CIDR subnet notation, “auto” or “none”) [default=auto]? none
INFO[06-15|14:10:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0
INFO[06-15|14:10:43] Sending request to LXD                   etag=etag method=PUT url=http://unix.socket/1.0
INFO[06-15|14:10:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/storage-pools/default
INFO[06-15|14:10:43] Sending request to LXD                   etag= method=POST url=http://unix.socket/1.0/storage-pools
INFO[06-15|14:10:43] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/networks/lxdbr0
INFO[06-15|14:10:43] Sending request to LXD                   etag= method=POST url=http://unix.socket/1.0/networks
INFO[06-15|14:10:45] Sending request to LXD                   etag= method=GET url=http://unix.socket/1.0/profiles/default
INFO[06-15|14:10:45] Sending request to LXD                   etag= method=PUT url=http://unix.socket/1.0/profiles/default
LXD has been successfully configured.
```
Maybe its viable to autogenerate [pressed](https://github.com/lxc/lxd/blob/master/doc/preseed.md) after lxd init run.
Also, is it ok to just?
```
lxd init <<EOF 
> yes
> default
> btrfs
> yes
> no
> yes
> no
> yes
> yes
> lxdbr0
> auto
> none
EOF
```
Some example: When I edit profile yaml in vim it helps me tab the right way, beside lxc reports miss spaces and style error in yaml configuration, and I simply can't save wrong configuration. But when I preseed to lxd init with EOF, basically it just ignore wrong lines, am I right?

**Comment by stgraber** [comment]
The LXD init questions won't randomly change on any given install, so sending the answers on stdin should work fine. Do note that we may however add, remove, change or re-order them between releases so that may be a problem after upgrade.

I've not played with preseeding much myself, but I'd be surprised if bad input was accepted without an error. However one thing that LXD will always silently ignore is extra keys. That's because of the way yaml and json is parsed into structs in Go.

Basically any key in the yaml or json document which isn't in the Go struct is simply ignored by the parser without ever reporting an error that we could forward to the user.
